mautic/mautic
# Fix package release (add deleted_files.txt)

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:

Successor of #10226 
Fixes #10195 

**This PR also bumps the minimum Mautic version to 3.2.0, as that version introduced PHP 7.4 support which is needed for Mautic 4.**

The deleted_files.txt is not included in the update.zip file. It means the legacy files are not deleted.



#### Steps to test this PR:
1. Install a Mautic 3.2.0 instance locally with the following commands:
```
mkdir mautic3test && cd mautic3test
ddev config --webserver-type=apache-fpm --php-version=7.4 && ddev start
wget https://github.com/mautic/mautic/releases/download/3.2.0/3.2.0.zip
unzip -q 3.2.0.zip
ddev exec php bin/console mautic:install https://mautic3test.ddev.site \
        --mailer_from_name="DDEV" --mailer_from_email="mautic@ddev.local" \
        --mailer_transport="smtp" --mailer_host="localhost" --mailer_port="1025" \
	--db_driver="pdo_mysql" --db_host="db" --db_port="3306" --db_name="db" --db_user="db" --db_password="db" \
        --db_backup_tables="false" --admin_email="admin@mautic.local" --admin_password="mautic"
```

2. Check if you can open the Mautic instance correctly
3. Copy [this file](https://fits4all.nl/4.0.0-beta-update.zip) to the root folder of the Mautic 3 installation
4. Run `ddev exec php bin/console mautic:update:apply --update-package=4.0.0-beta-update.zip`
5. Run `ddev exec php bin/console mautic:update:apply --finish`
6. You should now be running Mautic 4
